components/Hello.tsx

```diff
@@ -0,0 +1,89 @@
+// components/Hello.tsx
+import React from 'react';
+import {Button, StyleSheet, Text, View} from 'react-native';
+
+export interface Props {
+  name: string;
+  enthusiasmLevel?: number;
+}
+
+interface State {
+  enthusiasmLevel: number;
+}
+
+export class Hello extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    if ((props.enthusiasmLevel || 0) <= 0) {
+      //throw new Error('You could be a little more enthusiastic. :D');
+    }
+
+    this.state = {
+      enthusiasmLevel: props.enthusiasmLevel || 1,
+    };
+  }
+
+  onIncrement = () =>
+    this.setState({enthusiasmLevel: this.state.enthusiasmLevel + 1});
+  onDecrement = () => {
+    if ((this.state.enthusiasmLevel || 0) > 0)
+        this.setState({enthusiasmLevel: this.state.enthusiasmLevel - 1});
+    }
+  getExclamationMarks = (numChars: number) => Array(numChars + 1).join('!');
```

Comment:
Pelo que eu vi você está usando esse método enviando `this.state. enthusiasmLevel`. Só que esse método tem acesso a esse state. Poderia pegar direto dentro dele. O que acha?

Reply:
Alan, muito obrigado pelas dicas! Fiz todas as alterações conforme você sugeriu, acho que agora está melhor.